File: libs/document-processor-next/src/__tests__/models/test-event-channel.ts
import { z } from 'zod';
import { withTypeBlueId } from '@blue-labs/language';

import { channelContractBaseSchema } from '../../model/shared/index.js';

export const testEventChannelSchema = withTypeBlueId('TestEventChannel')(
  channelContractBaseSchema.extend({
    eventType: z.string(),
  })
);

export type TestEventChannel = z.infer<typeof testEventChannelSchema>;

File: references/java-sdk/src/test/java/blue/language/processor/model/TestEventChannel.java
package blue.language.processor.model;

import blue.language.model.TypeBlueId;
import blue.language.processor.model.ChannelContract;

@TypeBlueId("TestEventChannel")
public class TestEventChannel extends ChannelContract {

    private String eventType;

    public String getEventType() {
        return eventType;
    }

    public void setEventType(String eventType) {
        this.eventType = eventType;
    }
}
